libs/libcommon/src/libcommon/viewer_utils/asset.py

```diff
@@ -65,6 +70,70 @@ class ImageSource(TypedDict):
     width: int
 
 
+def create_asset_file(
+    dataset: str,
+    config: str,
+    split: str,
+    row_idx: int,
+    column: str,
+    filename: str,
+    assets_base_url: str,
+    assets_directory: StrPath,
+    fn: partial,  # type: ignore
+    overwrite: bool = True,
+    # TODO: Once assets and cached-assets are migrated to S3, this parameter is no more needed
+    use_s3_storage: bool = False,
+    s3_client: Optional[S3Client] = None,
+    # TODO: Once assets and cached-assets are migrated to S3, the following parameters dont need to be optional
+    s3_bucket: Optional[str] = None,
+    s3_folder_name: Optional[str] = None,
+) -> str:
+    dir_path, url_dir_path = get_asset_dir_name(
+        dataset=dataset,
+        config=config,
+        split=split,
+        row_idx=row_idx,
+        column=column,
+        assets_directory=assets_directory,
+    )
+
+    if not use_s3_storage:
+        file_path = dir_path / filename
+        makedirs(dir_path, ASSET_DIR_MODE, exist_ok=True)
+    else:
+        payload = (
+            str(uuid4()),
```

Comment:
if we use uuid, no need for the rest of the payload, right? let's just set the prefix using the uuid.